--- matlab2tikz.m       2013-09-17 20:49:43.897143600 +0200
+++ matlab2tikz_M.m     2013-10-06 23:39:01.550293600 +0200
@@ -2934,16 +2934,28 @@

   xData = get(c(1), 'XData');
   yData = get(c(1), 'YData');
-
+  zData = get(c(1), 'ZData');
+
   step = 3;
   m = length(xData(1:step:end));   % number of arrows

-  XY = zeros(4,m);
+  if(isempty(zData))
+      XY = zeros(4,m);

-  XY(1,:) = xData(1:step:end);
-  XY(2,:) = yData(1:step:end);
-  XY(3,:) = xData(2:step:end);
-  XY(4,:) = yData(2:step:end);
+      XY(1,:) = xData(1:step:end);
+      XY(2,:) = yData(1:step:end);
+      XY(3,:) = xData(2:step:end);
+      XY(4,:) = yData(2:step:end);
+  else
+      XYZ = zeros(6,m);
+
+      XYZ(1,:) = xData(1:step:end);
+      XYZ(2,:) = yData(1:step:end);
+      XYZ(3,:) = zData(1:step:end);
+      XYZ(4,:) = xData(2:step:end);
+      XYZ(5,:) = yData(2:step:end);
+      XYZ(6,:) = zData(2:step:end);
+  end
   % - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   % gather the arrow options
   showArrowHead = get(h, 'ShowArrowHead');
@@ -2980,10 +2992,19 @@
                                      ['{', arrowOptions, '}']);
   % - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   % return the vector field code
-  str = [str, ...
-         sprintf(['\\addplot [arrow',num2str(m2t.quiverId), '] ', ...
-                  'coordinates{(',m2t.ff,',',m2t.ff,') (',m2t.ff,',',m2t.ff,')};\n'],...
-                  XY)];
+  if(isempty(zData))
+      % quiver() 2D plot
+      str = [str, ...
+             sprintf(['\\addplot [arrow',num2str(m2t.quiverId), '] ', ...
+                      'coordinates{(',m2t.ff,',',m2t.ff,') (',m2t.ff,',',m2t.ff,')};\n'],...
+                      XY)];
+  else
+      % quiver3() 3D plot
+      str = [str, ...
+             sprintf(['\\addplot3 [arrow',num2str(m2t.quiverId), '] ', ...
+                      'coordinates{(',m2t.ff,',',m2t.ff,',',m2t.ff,') (',m2t.ff,',',m2t.ff,',',m2t.ff,')};\n'],...
+                      XYZ)];
+  end
   % - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   return;
 end
